projects/dex-ui/src/utils/logger.ts

```diff
@@ -0,0 +1,75 @@
+const DEBUG = import.meta.env.VITE_DEBUG;
+
+class Logger {
+  static debugConfig: Record<string, boolean> = {
+    DEFAULT: false
+  };
+  static modules = new Map<string, Logger>();
+  name: string;
+
+  constructor(name?: string) {
```

Comment:
well, it is accepting the module name in the constructor, it's just using a factory, with caching, instead of asking the user to create an instance on every page, and tracking that instance so it doesn't get re-created on every render, etc...

Reply:
I think there's room for improvement but it's not a priority right now. It's fine.